src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clon = [];
+  let newState = {
+    ...state,
+  };
+
+  for (const action of actions) {
+    newState = {
+      ...newState,
+    };
+
```

Comment:
you could make this copy after the switch - during `.push` - see below
if you, you won't need this here
```suggestion
```